### ISSUE TITLE
It's not possible to modify a aws_acm_certificate that's in use. It needs lifecycle.create_before_destroy

### ISSUE THREAD

**Issue by kitsunde** [terraform-providers/terraform-provider-aws#4712]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.7
+ provider.aws v1.21.0

### Affected Resource(s)

* aws_acm_certificate

### Terraform Configuration Files

```hcl
provider "aws" {
  region = "us-east-1"
}

variable "domain" {}

variable "zone_name" {}

resource "aws_acm_certificate" "cert" {
  domain_name       = "${var.domain}"
  validation_method = "DNS"
}

data "aws_route53_zone" "zone" {
  name = "uatdomains.com."
}

resource "aws_route53_record" "records" {
  name = "${lookup(aws_acm_certificate.cert.domain_validation_options[0], "resource_record_name")}"
  type = "${lookup(aws_acm_certificate.cert.domain_validation_options[0], "resource_record_type")}"

  zone_id = "${data.aws_route53_zone.zone.id}"

  records = [
    "${lookup(aws_acm_certificate.cert.domain_validation_options[0], "resource_record_value")}",
  ]

  ttl = 60
}

resource "aws_acm_certificate_validation" "cert_validation" {
  certificate_arn = "${aws_acm_certificate.cert.arn}"

  validation_record_fqdns = [
    "${aws_route53_record.records.*.fqdn}",
  ]
}

resource "aws_elb" "test" {
  availability_zones = ["us-east-1a"]

  "listener" {
    instance_port      = 443
    instance_protocol  = "https"
    lb_port            = 443
    lb_protocol        = "https"
    ssl_certificate_id = "${aws_acm_certificate.cert.arn}"
  }
}
```

### Execution Output

```
$ terraform apply
aws_acm_certificate.cert: Refreshing state... (ID: arn:aws:acm:us-east-1:178284945954:cert...e/0bc605bf-d8a8-4bd7-810b-9623b2d7cb9b)
data.aws_route53_zone.zone: Refreshing state...
aws_elb.test: Refreshing state... (ID: tf-lb-20180531093408871400000001)
aws_route53_record.records: Refreshing state... (ID: Z1OC43VLHN0C3T__7df6b3ab30e3cba676e7ffa...dify-certificate.uatdomains.com._CNAME)
aws_acm_certificate_validation.cert_validation: Refreshing state... (ID: 2018-05-31 09:30:05 +0000 UTC)

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place
-/+ destroy and then create replacement

Terraform will perform the following actions:

-/+ aws_acm_certificate.cert (new resource required)
      id:                                      "arn:aws:acm:us-east-1:178284945954:certificate/0bc605bf-d8a8-4bd7-810b-9623b2d7cb9b" => <computed> (forces new resource)
      arn:                                     "arn:aws:acm:us-east-1:178284945954:certificate/0bc605bf-d8a8-4bd7-810b-9623b2d7cb9b" => <computed>
      domain_name:                             "modify-certificate.uatdomains.com" => "modify-certificate-1.uatdomains.com" (forces new resource)
      domain_validation_options.#:             "1" => <computed>
      validation_emails.#:                     "0" => <computed>
      validation_method:                       "DNS" => "DNS"

-/+ aws_acm_certificate_validation.cert_validation (new resource required)
      id:                                      "2018-05-31 09:30:05 +0000 UTC" => <computed> (forces new resource)
      certificate_arn:                         "arn:aws:acm:us-east-1:178284945954:certificate/0bc605bf-d8a8-4bd7-810b-9623b2d7cb9b" => "${aws_acm_certificate.cert.arn}" (forces new resource)
      validation_record_fqdns.#:               "1" => <computed> (forces new resource)

  ~ aws_elb.test
      listener.4170238750.instance_port:       "443" => "0"
      listener.4170238750.instance_protocol:   "https" => ""
      listener.4170238750.lb_port:             "443" => "0"
      listener.4170238750.lb_protocol:         "https" => ""
      listener.4170238750.ssl_certificate_id:  "arn:aws:acm:us-east-1:178284945954:certificate/0bc605bf-d8a8-4bd7-810b-9623b2d7cb9b" => ""
      listener.~3078868615.instance_port:      "" => "443"
      listener.~3078868615.instance_protocol:  "" => "https"
      listener.~3078868615.lb_port:            "" => "443"
      listener.~3078868615.lb_protocol:        "" => "https"
      listener.~3078868615.ssl_certificate_id: "" => "${aws_acm_certificate.cert.arn}"

-/+ aws_route53_record.records (new resource required)
      id:                                      "Z1OC43VLHN0C3T__7df6b3ab30e3cba676e7ffac25023845.modify-certificate.uatdomains.com._CNAME" => <computed> (forces new resource)
      allow_overwrite:                         "true" => "true"
      fqdn:                                    "_7df6b3ab30e3cba676e7ffac25023845.modify-certificate.uatdomains.com" => <computed>
      name:                                    "_7df6b3ab30e3cba676e7ffac25023845.modify-certificate.uatdomains.com" => "${lookup(aws_acm_certificate.cert.domain_validation_options[0], \"resource_record_name\")}" (forces new resource)
      records.#:                               "1" => <computed>
      ttl:                                     "60" => "60"
      type:                                    "CNAME" => "${lookup(aws_acm_certificate.cert.domain_validation_options[0], \"resource_record_type\")}"
      zone_id:                                 "Z1OC43VLHN0C3T" => "Z1OC43VLHN0C3T"


Plan: 3 to add, 1 to change, 3 to destroy.
```

### Expected Behavior

It should be possible to update a certificate. Can we set `create_before_destroy` by default since it's pretty surprising and certificates are free?

### Actual Behavior

The certificate can't be destroyed because it's in use.

```
Error: Error applying plan:

1 error(s) occurred:

* module.terraform-redirect.module.certificate.aws_acm_certificate.cert (destroy): 1 error(s) occurred:

* aws_acm_certificate.cert: Error deleting certificate: ResourceInUseException: Certificate arn:aws:acm:us-east-1:178284945954:certificate/02a83993-0289-4ad6-8d24-7c068dc000d0 in account 178284945954 is in use.
	status code: 400, request id: 7cda32e7-64ab-11e8-bf3b-fd4416fbb2cf
```

### Steps to Reproduce

1. `terraform apply -var=domain=modify-certificate.uatdomains.com -var=domain=uatdomains.com.`
2. `terraform apply -var=domain=modify-certificate-fail.uatdomains.com -var=domain=uatdomains.com.`

### Important Factoids

The LB up there is in EC2 Classic, but the same thing happens on CloudFront or anything that's using the cert.

**Comment by bflad** [comment]
> Can we set create_before_destroy by default

I do not believe this is possible currently. Terraform core controls the ordering behavior.

It does seem like this might be something valid to add to the resource documentation though, especially for newer operators who maybe have not seen the lifecycle configuration before.

**Comment by bflad** [comment]
Documentation update has been merged and will release with version 1.25.0 of the AWS provider, likely middle of this week.
